lib/alertwildfire_API.py

```diff
@@ -0,0 +1,109 @@
+# Copyright 2018 The Fuego Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+import os
+import sys
+fuegoRoot = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+sys.path.insert(0, os.path.join(fuegoRoot, 'lib'))
+sys.path.insert(0, fuegoRoot)
+import settings
+settings.fuegoRoot = fuegoRoot
+import datetime
+import requests
+import time
+import img_archive
+
+import urllib.parse as urlp
+baseApiUrl =urlp.ParseResult(scheme='https', netloc='data.alertwildfire.org', path='/api/firecams/v0', params='', query='', fragment='')#initialize
+
+
+
+if settings.alertwildfirekey == 'do not put real key in files in open source git repo':
+    logging.warning('Please update settings.alertwildfirekey with the key provided to you')
+
+
+def get_all_camera_info():
+    headers = {'X-Api-Key': settings.alertwildfirekey}
+    urlParts = baseApiUrl._replace(path = baseApiUrl.path+"/cameras")
+    url = urlp.urlunparse(urlParts)
+    response = requests.get(url, headers=headers)
```

Comment:
I think it would be cleaner to have a function getApiUrl(endpoint, queryParams=None).
Then, we could hide baseApiUrl._replace and unparse inside it instead of replicating it everywhere.
We could even go further and make function called invokeApi(endpoint, queryParams, stream = None/False) that returns the response, and then avoid replicating the code headers key info as well.